cpp/include/algorithm/sorting/README.md

```diff
@@ -0,0 +1,204 @@
+_O(N * log(N)_ | _O(1)_
+
+Where N is the number of elements.
+
+## 6. Counting Sort
+An integer sorting algorithm that operates by counting the number of objects that have each distinct key value, and using arithmetic on those counts to determine the positions of each key value in the output sequence. It is a stable and comparision based algorithm. 
+
+### Usage
+``` c++
+vector<int> Arr{1, 8, 5, 12, 3};
+couting_sort(Arr, -1, true)
+```
+### Complexity
+
+Time    | Space
+:--------:|:-------------------:
+_O(N + R)_ | _O(N + K)_
+
+Where N is the number of elements, R is the range of input and K the size of the frequency array.
+
+## 7. Bucket Sort
+A sorting algorithm that works by dividing elements into different buckets, and then sorting the buckets individualy. Each bucket is then sorted individually by using a different sorting algorithm. Insertion sort is used in this implementation, but other algorithms can be used.
+
+### Usage
+``` c++
+vector<int> Arr{1, 8, 5, 12, 3};
+bucket_sort(Arr, -1, true)
+```
+### Complexity
+
+Time    | Space
+:--------:|:-------------------:
+_O(N + N^2 / K+K)_ | _O(N)_
+
+Where N is the number of elements, and K is the number of buckets
+
+## 8. Radix Sort
+Least significant digit radix sort is an efficient, non-comparative, integer sorting algorithm that sorts data with integer keys by grouping keys by the individual digits which share the same significant position and value. Note that this implementation of radix sort can be used to sort positive integers only.
+
+### Usage
+``` c++
+vector<int> Arr{1, 8, 5, 12, 3};
+radix_sort(Arr, -1, true)
+```
+### Complexity
+
+Time    | Space
+:--------:|:-------------------:
+_O(N)_ | _O(N)_
+
+Where N is the number of keys
+
+## 9. Heap Sort
+A comparison-based sorting algorithm that uses an array-implemented heap to sort a list of integers.
+
+### Usage
+``` c++
+vector<int> Arr{1, 8, 5, 12, 3};
+heap_sort(Arr, -1, true)
+```
+### Complexity
+
+Time    | Space
+:--------:|:-------------------:
+_O(N * log(N))_ | _O(1)_
+
+Where N is the number of elements
+
+## 10. Shell Sort
+An in-place comparison sort which starts by sorting pairs of elements far apart from each other, then progressively reducing the gap between elements to be compared. The time complexity depends on the gap sequence, which is _floor(N / 2^K)_ in this implementation.
+
+### Usage
+``` c++
+vector<int> Arr{1, 8, 5, 12, 3};
+shell_sort(Arr, -1, true)
+```
+### Complexity
+
+Time    | Space
+:--------:|:-------------------:
+_O(N^2)_ | _O(1)_
+
+Where N is the number of elements
+
+## 11. Comb Sort
+A sorting algorithm designed to improve upon bubble sort by by eliminating small values that start near the end of the list.
+	
+### Usage
+``` c++
+vector<int> Arr{1, 8, 5, 12, 3};
+comb_sort(Arr, -1, true)
+```
+### Complexity
+
+Time    | Space
+:--------:|:-------------------:
+_O(N^2 / 2^p)_ | _O(1)_
+
+Where N is the number of elements, and p is the number of increments
```

Comment:
Add newline.

Reply:
I'm assuming here you mean one additional whitespace? Or did you mean \n?